File: toolrental/src/store/models/Product.ts
import Category from "./Category";

export default interface IProduct {
  id: number,
  name: string,
  shortDescription: string,
  longDescription: string,
  rentalPrice: number,
  isInStock: boolean,
  imageSrc: string,
  category: Category | null,
}

export default class Product implements IProduct{
  constructor(options: IProduct) {
    Object.assign(this, options);

    this.category = options.category == null ? null : new Category(options.category);
  }

  id = 0;
  name = "";
  shortDescription = "";
  longDescription = "";
  rentalPrice = 0;
  isInStock = false;
  imageSrc = ""
  category: Category | null = null;
}

File: ToolRental.API/ToolRental.API/Models/Response/Product.cs
namespace ToolRental.API.Models.Response
{
	public class Product
	{
		public uint Id { get; set; }
		public string? Name { get; set; }
		public string? ShortDescription { get; set; }
		public string? LongDescription { get; set; }
		public uint RentalPrice { get; set; }
		public bool IsInStock { get; set; }
		public string? ImageSrc { get; set; }
		public Category? Category { get; set; }
	}
}
